app/Http/Controllers/Admin/UserController.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+namespace App\Http\Controllers\Admin;
+
+use Illuminate\Http\Request;
+use App\Http\Controllers\Controller;
+use App\Models\User;
+
+class UserController extends Controller
+{
+
+    protected $user;
+    protected $perPage;
+
+    public function __construct()
+    {
+        $this->middleware('auth:admin');
+        $this->perPage = config('pagination.per_page');
+    }
+
+    private function getUser($id){
+        $user = User::find($id);
+        if(!$user){
+            return back()
+            ->withErrors(['msg', 'Has a error!']);
+        }
+        return $user;
+    }
+
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index(Request $request)
+    {
+        if($request->has('search')) {
+            $searchTerm = $request->get('search');
+            $users = User::search($searchTerm)->paginate();
+            return view('admin.user.index')
+                ->with(['users' => $users, 'search' => $searchTerm])
+                ->withMessage('Search result');
+        }
+        $users = User::paginate($this->perPage);
```

Comment:
cái này thầy viết lên trên luôn ạ. 
 $users = User::search($searchTerm)->paginate($this->perPage);